### PR TITLE
Allow for building roller coasters with all 4 curve types

### PR DESCRIPTION
**Pull request by Gymnasiast**
Requested by @RealSteel89 for the Spinning Roller Coaster.

Tested with several ride types, including the Multi-dimension.

In future PRs, we can use the extra space to e.g. allow for building long-base and small flat-to-60-degree slopes.

afbeelding
afbeelding
afbeelding

### CODE REVIEW

**Comment by karst** [comment]
This does make the construction menu quite a bit wider.
I would like to bring up #17572, that it might be a good idea after this PR to look into reworking the entire menu.
I think people might not enjoy the change to start, but will appreciate it once we have gone through with it.
This way we can choose between the different kinds of switch between flat/steep as well.

**Comment by Realsteel89** [comment]
That was the idea behind expanding the ui later so that you can unlock 1 tile flat to steep and 3 tiles without a cheat as well as the 4 curve sizes. This appeared for the first time usable on my previous "Maurer" track. Later you could expand the ui further, e.g. change menu of brakes or booster types (e.g. Giga) or also types of lift sprites, i.e. chain or friction wheel drive. I am currently able to render giga, mini and" Maurer" track addings.

File: src/openrct2-ui/windows/RideConstruction.cpp [hunk 1]
```diff
@@ -1611,33 +1611,15 @@ class RideConstructionWindow final : public Window
...
         if (IsTrackEnabled(TRACK_CURVE))
         {
             widgets[WIDX_LEFT_CURVE].type = WindowWidgetType::FlatBtn;
             widgets[WIDX_RIGHT_CURVE].type = WindowWidgetType::FlatBtn;
-            widgets[WIDX_LEFT_CURVE_SMALL].left = 6;
-            widgets[WIDX_LEFT_CURVE_SMALL].right = 27;
-            widgets[WIDX_LEFT_CURVE_SMALL].image = ImageId(SPR_RIDE_CONSTRUCTION_LEFT_CURVE_SMALL);
-            widgets[WIDX_RIGHT_CURVE_SMALL].left = 138;
-            widgets[WIDX_RIGHT_CURVE_SMALL].right = 159;
-            widgets[WIDX_RIGHT_CURVE_SMALL].image = ImageId(SPR_RIDE_CONSTRUCTION_RIGHT_CURVE_SMALL);
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/RideConstruction.cpp:1640]: I'm not sure I follow this part, what it was doing was to replace the small curve icons by maybe not a small curve when the ride didn't had one, so as to not increase the width of the coaster?

File: src/openrct2-ui/windows/RideConstruction.cpp [hunk 2]
```diff
@@ -1701,19 +1683,17 @@ class RideConstructionWindow final : public Window
...
             widgets[WIDX_CHAIN_LIFT].type = WindowWidgetType::FlatBtn;
-            x = 9;
         }
         else
         {
             widgets[WIDX_CHAIN_LIFT].type = WindowWidgetType::Empty;
-            x = 23;
         }
 
+        int32_t x = 45;
```
Comment (tupaschoal) [inline src/openrct2-ui/windows/RideConstruction.cpp:1696]: Not sure I get this change either :'(

Comment (Realsteel89) [reply src/openrct2-ui/windows/RideConstruction.cpp:1696]: it makes the ui wider and makes it easier to assign the very narrow curves to a symbol by inserting a new ui if the tracks have it. currently only my maurer has the 4 different curve radii active

File: src/openrct2-ui/windows/RideConstruction.cpp (hunk 1 above)
Comment (Gymnasiast) [reply src/openrct2-ui/windows/RideConstruction.cpp:1640]: It previously only had space for 7 buttons side by side (3 types of curve, both left and right, plus straight on), so it had to rearrange the placement. Now that the UI is wide enough to display all 9 buttons necessary, it doesn’t need to change positions any more.

**Review by tupaschoal — APPROVED** [review]
Seems ok, though I haven't tested.

**Review by karst — APPROVED** [review]
All seems to work properly, everything is alligned correctly to the centre. Excited to see this window grow from here.

**Comment by Realsteel89** [comment]
nice that is merged but nobody had tested it with an track that has all the 4 curves active because maurer was the first track they can do it
grafik
that i got on an earlier version no idea if it was fixed than^^

**Comment by cheweytoo** [comment]
Nice!

This would be the opportunity to also make the coaster's buildable slopes more visible and discoverable, like e.g.:

wide_build

(with appropriate "browning out" of those that cannot be attached to the current piece of course)

**Pull request merged**